tests/unit/via/security_test.py

```diff
@@ -0,0 +1,46 @@
+from unittest.mock import sentinel
+
+import pytest
+from h_matchers import Any
+from pyramid.security import Allowed, Denied
+
+from via.security import ViaSecurityPolicy
+
+
+class TestViaSecurityPolicy:
+    def test_permits_when_theres_no_authorization_header(self, pyramid_request):
+        result = ViaSecurityPolicy().permits(pyramid_request, sentinel.context, "api")
+
+        assert result == Any.instance_of(Denied).with_attrs(
+            {"msg": "No Authorization header"}
+        )
```

Comment:
Note that we're asserting about the message strings that the security policy passes to `Denied()`. This is to defend against bugs in the tests. For example a test that the security policy denies a request with an invalid JWT might have a bug in the test code such that it doesn't actually put _any_ JWT in the request. The security policy would still return `Denied()` (because there's no JWT) and the test would pass. The security policy could actually be failing to deny requests that do contain an invalid JWT and the test would still be passing. Having the tests assert about the denied reason defends against this.

Similar to asserting about exception messages when testing exception raising